src/os/file.go

```diff
@@ -181,7 +181,7 @@ func (f *File) SyscallConn() (syscall.RawConn, error) {
 
 // Fd returns the file handle referencing the open file.
 func (f *File) Fd() uintptr {
-	panic("unimplemented: os.file.Fd()")
+	return f.handle.Fd()
```

Comment:
Here is an example:

```suggestion
	handle, ok := f.handle.(interface {
		Fd() uintptr
	})
	if ok {
		return handle.Fd()
	}
	return 0
```

(This inline interface is perhaps a bit ugly though)